bin/pycbc_compress_bank

```diff
@@ -0,0 +1,199 @@
+parser.add_argument("--interpolation", type=str, default="linear",
+                help="If check-compression, the interpolation to use for "
+                     "decompressing the waveforms. Options are 'linear', or "
+                     "any interpolation recognized by scipy's interp1d kind "
+                     "argument. Default is linear.")
+parser.add_argument("--verbose", action="store_true", default=False)
+
+
+args = parser.parse_args()
+
+pycbc.init_logging(args.verbose)
+
+if args.fixed_sample_points and args.compression_algorithm != 'mchirp':
+    raise ValueError("fixed-sample-points requires the mchirp algorithm")
+
+df = 1./args.segment_length
+fmin = args.low_frequency_cutoff
+# figure out the length of the full waveforms in the frequency domain
+N = int(args.sample_rate*args.segment_length/2 + 1)
+# scratch space
+scratch = FrequencySeries(numpy.zeros(N, dtype=numpy.complex128), delta_f=df)
+if args.check_compression is not None:
+    decomp_scratch = FrequencySeries(numpy.zeros(N, dtype=numpy.complex128),
+        delta_f=df)
+# we'll need the full frequency series
+allf = numpy.arange(N)*df
+
+# load the bank
+logging.info("loading bank")
+bank = waveform.FilterBank(args.bank_file, N, df, fmin, numpy.complex128,
+    approximant=args.approximant, out=scratch)
+templates = bank.table
+if args.tmplt_index is not None:
+    imin, imax = args.tmplt_index
+    templates = templates[imin:imax]
+else:
+    imin, imax = 0, templates.size
+
+# the index of the starting frequency in the full frequency series
+kmin = int(fmin*args.segment_length)
+
+# we'll use the SEOBNRv2 ringdown frequency as maximum frequencies for the
+# sample points
+logging.info("computing maximum frequencies")
+kmaxes = (args.segment_length*pnutils.get_freq('fSEOBNRv2RD',
+            templates.mass1, templates.mass2,
+            templates.spin1z, templates.spin2z)).astype(int)
+kmaxes[kmaxes > N] = N
+
+# generate output file
+# FIXME: this should be moved to a class or function
+logging.info("writing template info to output")
+output = h5py.File(args.output, 'w')
+output['mass1'] = templates['mass1']
+output['mass2'] = templates['mass2']
+output['spin1z'] = templates['spin1z']
+output['spin2z'] = templates['spin2z']
+output['template_hash'] = templates['template_hash']
+
+if args.fixed_sample_points:
+    logging.info("getting fixed sample points")
+    useidx = pnutils.mass1_mass2_to_mchirp_eta(
+        templates.mass1, templates.mass2)[0].argmin()
+    fixed_points = compress.mchirp_compression(templates.mass1[useidx],
+        templates.mass2[useidx], fmin, df*(kmaxes.max()-1),
+        min_seglen=args.t_pad, df_multiple=df)
+    output['compressed_waveforms/frequency'] = fixed_points
+else:
+    sample_points = None
+    
+# get the compressed sample points for each template
+logging.info("getting compressed amplitude and phase")
+for ii in range(imin, imax):
+    htilde = bank[ii]
+    tmplt = bank.table[ii]
+    kmax = kmaxes[ii]
```

Comment:
Any reason you aren't using the kmax that the template already calculates (and which might be different from SEOBNR RD?)